### PR TITLE
Updates timestamps to Millisecond Precision - Pre-Confirmations Decay

### PR DESCRIPTION
**Pull request by ckartik**
## Millisecond Timestamps
- This updates the timestamps to be encodings of Milliseconds from EPOCH rather than seconds
- NOTE: This may impact the consumers of block timestamps on L2, such as the Block Explorer
Screenshot 2024-03-04 at 11 42 32 AM

### CODE REVIEW

File: consensus/clique/clique.go [hunk 1]
```diff
@@ -255,7 +256,7 @@ func (c *Clique) verifyHeader(chain consensus.ChainHeaderReader, header *types.H
 	number := header.Number.Uint64()
 
 	// Don't waste time checking blocks from the future
-	if header.Time > uint64(time.Now().Unix()) {
+	if header.Time > uint64(time.Now().UnixMilli()) {
```
[the remark this answers is not in the document]
Comment (ckartik) [reply consensus/clique/clique.go:259]: As talked about on our call, the header.Time is set in Milisecond format below in the code, and the genesis file has been updated to ensure that the initial value for the header.Time is also in Milisecond format.

File: consensus/clique/clique.go [hunk 2]
```diff
@@ -657,7 +659,7 @@ func (c *Clique) Seal(chain consensus.ChainHeaderReader, block *types.Block, res
 		}
 	}
 	// Sweet, the protocol permits us to sign the block, wait for our time
-	delay := time.Unix(int64(header.Time), 0).Sub(time.Now()) // nolint: gosimple
+	delay := time.UnixMilli(int64(header.Time)).Sub(time.Now()) // nolint: gosimple
```
Comment (shaspitz) [inline consensus/clique/clique.go:662]: Does the `wiggle` variable below, which is added to `delay`, also need changing?

File: geth-poa/genesis.json [hunk 3]
```diff
@@ -20,7 +20,7 @@
       }
     },
     "nonce": "0x0",
-    "timestamp": "0x6546df6d",
+    "timestamp": "0x18E0A4E0D22",
```
Comment (shaspitz) [inline geth-poa/genesis.json:23]: 0x6546df6d * 1000 = `0x18b9cd8c1c8`, changing this genesis starting timestamp shouldn't matter tho right

Comment (ckartik) [reply geth-poa/genesis.json:23]: Yeah it doesn't matter

File: consensus/clique/clique.go (hunk 2 above)
Comment (ckartik) [reply consensus/clique/clique.go:662]: I've reduced it to a 100ms.

**Review by shaspitz — APPROVED** [review]

**Pull request merged**